validator/stateless_block_validator.go

```diff
@@ -307,11 +307,17 @@ func SetMachinePreimageResolver(ctx context.Context, mach *ArbitratorMachine, pr
 			if err != nil {
 				log.Error("Failed to deserialize DAS message", "err", err)
 			} else {
-				dasPreimage, err := das.Retrieve(ctx, seqMsg[40:])
+				keysetPreimage, err := das.KeysetFromHash(ctx, cert.KeysetHash[:])
+				if err != nil {
+					return fmt.Errorf("couldn't keyset from DAS %w", err)
+				}
+				dasPreimage, err := das.Retrieve(ctx, cert)
 				if err != nil {
 					return fmt.Errorf("couldn't retrieve message from DAS %w", err)
```

Comment:
Will this error if there's a bad signature? We should continue validation without the data if so. Just make sure to still include the keyset hash in the preimages list.

Reply:
This is designed to populate the preimage table, if at all possible.  So even if the signatures are bad, the preimage table will be filled in.  (This is one reason we generally don't consistency-check the values in intermediary functions like RPC client/server and DAS aggregator.)

Reply:
Returning an error here will stop validation though, which we don't want to do if there's an invalid DAS signature (the chain can continue in that case)

Reply:
Switched to logging an error and continuing here, except for the case where the cert is valid but the DAS fails to produce data.  In that case we have to return error and stop.